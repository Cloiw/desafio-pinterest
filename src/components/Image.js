import React from 'react';
import './styles/Body.css';



class Image extends React.Component{ 

  
    render() {
    
      return (
          <>
          <div className="image-div">
            <img  className="image-size" src={this.props.url} alt={this.props.url}/>
          <span className="image-span">...</span>
          </div>
        </>
    );
  }
}

export default Image