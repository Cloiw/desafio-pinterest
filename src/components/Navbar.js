import React from 'react';
import logo from '../img/logo.svg';
import './styles/Navbar.css';


class Navbar extends React.Component{ 
    render() {
        return (
          <>
          <nav>
          <div className="nav-container">
            <div className="nav-logo">
                <a href="/">
                    <img className="nav-logo-icon" src={logo} alt="logo"/> 
                </a> 
            </div>
            <div className="nav-search">

            </div>
            <div className="nav-home">
              <span>Inicio</span>
            </div>
            <div className="nav-follow">
            <span>Siguiendo</span>
            </div>
           
            </div>
          </nav>



          

        </>
    );
  }
}

export default Navbar;