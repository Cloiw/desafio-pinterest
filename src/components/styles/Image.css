.show-save-button{
    position: absolute;
    margin:5px 5px 0px 121.5px;
    z-index: 1;
}

.save-icon{
    border-radius: 8px;
}

.image-size{
    cursor:zoom-in;
    width: 230px;
    height: auto;
    border-radius:8px;
}

.image-div{
    margin:4px;
    padding:6px 6px 10px 6px;
}

.flex-dots{
    display:flex;
    justify-content: flex-end;
    height: 100%;
}

.image-dots{
    margin-top:2px;
    width:16px;
    height:14px;
    padding:4px;
    color:#8d8d8d;
    font-weight: bolder;
}
.hide-save-button{
    display:none;
}
.image-dots:hover{
    border-radius: 100px;
    background-color:  rgba(0, 0, 0, .05);
}
.image-div:hover{
    background-color: #f2f2f2;
    border-radius: 8px;
}
.image-div:hover > img{
    -webkit-filter: brightness(60%);
    filter: brightness(60%);
    z-index: 0;
}