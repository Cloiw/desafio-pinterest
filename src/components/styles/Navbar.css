.nav-container{
  height: 100%;
  display: flex;
  align-items: center;
  padding-left:12px;
}

nav{
  z-index:2;
  position: fixed;
  background-color: white;
  width: 100%;
  height: 64px;
  top: 0;
  color:#333;
  font-weight: bold;
  font-size: 16px;
  border-bottom: 1px solid rgba(0, 0, 0, .05);
}

.nav-home,.nav-follow{
  padding:8px;
  color:#8E8E8E;
  flex:1;
}

.nav-search{
  height: 40px;
  background-color:  rgba(0, 0, 0, .05);
  flex:4;
  margin-left: 16px;
  margin-right: 16px;
  border-radius: 8px;
}

.nav-logo{
  display: flex;
  align-items: center;
  justify-content: center;
  padding:4px;
  height: 40px;
  width: 40px;
}

.nav-logo:hover{
  border-radius: 100px;
  background-color:  rgba(0, 0, 0, .05);
}

.nav-logo-icon,.nav-logo a{
  height: 28px;
  width: 28px;
}