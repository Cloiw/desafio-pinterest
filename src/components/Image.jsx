import React from 'react';
import './styles/Image.css';
import dotsIcon from '../img/dots_icon.png';
import saveIcon from '../img/save_icon.jpg';

class Image extends React.Component {
  constructor() {
    super();
    this.state = { isHover: false };
    this.handleHover = this.handleHover.bind(this);
  }

  handleHover = () => {
    this.setState((prevState) => (
      { isHover: !prevState.isHover }
    ));
  };

  render() {
    return (
      <>
        <div onMouseEnter={() => this.handleHover()} onMouseLeave={() => this.handleHover()} className="image-div">
          <button type="button" className={this.state.isHover ? 'show-save-button' : 'hide-save-button'}>
            <img className="save-icon" src={saveIcon} alt="Guardar" />
          </button>
          <img className="image-size" src={this.props.url} alt={this.props.url} />
          <div className="flex-dots">
            <img className="image-dots" src={dotsIcon} alt="More" />
          </div>
        </div>
      </>
    );
  }
}

export default Image;
