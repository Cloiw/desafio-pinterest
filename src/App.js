import React from 'react';
import Navbar from './components/Navbar'
import Body from './components/Body'

import './App.css';

function App() {
  return (
    
      <div>
        <header>
          <Navbar/>
        </header>
        <Body/>
      </div>
  )
}

export default App;
