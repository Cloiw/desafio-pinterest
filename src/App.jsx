import React from 'react';
import Body from './components/Body';
import Navbar from './components/Navbar';
import './App.css';

function App() {
  return (
    <div>
      <header>
        <Navbar />
      </header>
      <Body />
    </div>
  );
}

export default App;
