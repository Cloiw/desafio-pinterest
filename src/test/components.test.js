import Image from '../components/Image'
import Navbar from '../components/Navbar'
import App from '../App'
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import React from 'react';
import Enzyme from 'enzyme';
import {shallow , mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });


it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
  <BrowserRouter>
  <App />
  </BrowserRouter>,
  div
  );
  ReactDOM.unmountComponentAtNode(div);
})

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
  <BrowserRouter>
  <Image />
  </BrowserRouter>,
  div
  );
  ReactDOM.unmountComponentAtNode(div);
})


it('se renderea correctamente el componente Navbar', () => {  
  const component = shallow(<Navbar />);
  expect(toJson(component)).toMatchSnapshot();
});