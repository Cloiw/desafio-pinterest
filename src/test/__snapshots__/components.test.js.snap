// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`se renderea correctamente el componente Navbar 1`] = `
<Fragment>
  <header>
    <nav>
      <div
        className="nav-container"
      >
        <div
          className="nav-logo"
        >
          <a
            href="/"
          >
            <img
              alt="logo"
              className="nav-logo-icon"
              src="logo.svg"
            />
          </a>
        </div>
        <div
          className="nav-search"
        />
        <div
          className="nav-home"
        >
          <span>
            Inicio
          </span>
        </div>
        <div
          className="nav-follow"
        >
          <span>
            Siguiendo
          </span>
        </div>
      </div>
    </nav>
  </header>
</Fragment>
`;
